anderson_erik_lidar_analysis.py

```diff
@@ -0,0 +1,262 @@
+
+    # Calculate zonal stats for each plot
+    chm_stats = rs.zonal_stats(
+        plots_gdf, chm_path,
+        stats=['mean', 'max'],
+        geojson_out = True, nodata=0
+    )
+    chm_stat_gdf = gpd.GeoDataFrame.from_features(chm_stats)
+    chm_stat_gdf = chm_stat_gdf.rename(
+        columns={'max': 'lidar_max', 'mean': 'lidar_mean'})
+    
+    if prepend_string:
+        chm_stat_gdf[prepend_col] = prepend_string + chm_stat_gdf['ID']
+    
+    return chm_stat_gdf
+
+
+# In[17]:
+
+
+def _calc_insitu_height_stats(insitu_path):
+    """
+    Caculate insitu max and mean tree height
+
+    Parameters
+    ----------
+    insitu_path: str
+      File path to insitu data
+
+    Returns
+    -------
+    GeoDataFrame
+      Plot data with stats by plot id
+    """ 
+    insitu_df = (
+        pd.read_csv(insitu_path)
+        .groupby('plotid')['stemheight']
+        .agg(['max', 'mean'])
+        .rename(columns={'max': 'insitu_max', 'mean': 'insitu_mean'})
+    )
+    df = gpd.GeoDataFrame(insitu_df)
+
+    return df
+
+
+# In[18]:
+
+
+def calc_height_stats(plots_path, chm_path, insitu_path, 
+                      id_col="Plot_ID", prepend_string=None):
+  """
+  Merge insitu and LiDAR data
+
+  Parameters
+  ----------
+  plots_path: str
+    File path to plot locations (point locations)
+  chm_path: str
+    File path to LiDAR file
+  insitu_path: str
+    File path to insitu data
+  id_col: str
+    Column name to join data on
+
+  Returns
+  -------
+  GeoDataFrame
+    Merged LiDAR and insitu data
+  """
+  
+  chm_stats_gdf = _calc_lidar_chm_stats(plots_path, chm_path, 
+                                        prepend_string=prepend_string,
+                                        prepend_col=id_col)
+  # Load and merge insitu data
+  insitu_height_df = _calc_insitu_height_stats(insitu_path)
+  
+  stat_df = (
+      chm_stats_gdf
+      .merge(insitu_height_df, left_on=id_col, right_index=True)
+  )
+  return stat_df
+
+
+# In[19]:
+
+
+def plot_comparison(df):
+    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12,4))
+    ax1.scatter(
+        df['lidar_mean'], df['insitu_mean']
+    )
+    ax1.set(xlim=(0, 30), ylim=(0, 30), aspect='equal')
+    ax1.set_title('Mean')
+    sns.regplot(x='lidar_mean', y='insitu_mean', 
+                data=df,
+                color='purple',
+                ax=ax1)
+    ax1.plot((0, 1), (0, 1), transform=ax1.transAxes, ls='--', c='k')
+    ax2.scatter(df['lidar_max'], df['insitu_max'])
+    ax2.set(xlim=(0, 30), ylim=(0, 30), aspect='equal')
+    ax2.set_title('Maximum')
+    sns.regplot(x='lidar_max', y='insitu_max', 
+                data=df,
+                color='purple',
+                ax=ax2)
+    ax2.plot((0, 1), (0, 1), transform=ax2.transAxes, ls='--', c='k')
+    plt.suptitle('Comparison of LiDAR and insitu height measurements at the \n'
+                'NEON San Joaquin Experimental Range (SJER) field site')
+    plt.tight_layout()
```

Comment:
You could add more # comments to clarify your plot settings

Reply:
will do, thanks!